type script/unAuthorized ~@Error.ts

```diff
@@ -0,0 +1,28 @@
+import { Inject, Optional } from "@angular/core";
+import { LocalStorageService } from "@core/services/local-storage.service";
+import { ModalService } from "src/app/shared/services/modal.service";
+import { ModalConstants } from "src/app/utils/modals-constants";
+
+export class UnAuthorizedError {
+
+    constructor(@Optional() @Inject(ModalService) private _modalService: ModalService,
+    @Optional() @Inject(LocalStorageService) private _localStorageService: LocalStorageService) {
+        this.handleUnauthorizedError();
+    }
+
+    async handleUnauthorizedError() {
+        console.log('inside handle unauthorized error');
+
+        let modalElement = document.getElementById('confirmation-session-expire') as HTMLElement;
+        if (modalElement) {
+            return;
+        }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider using Angular modal tracking mechanism</b></div>
<div id="fix">

The check for existing modal using `document.getElementById()` might not be reliable in Angular applications. Consider using Angular's view querying mechanisms or the `ModalService` to track active modals.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -13,10 +13,7 @@
     async handleUnauthorizedError() {
         console.log('inside handle unauthorized error');

 -        let modalElement = document.getElementById('confirmation-session-expire') as HTMLElement;
 -        if (modalElement) {
 -            return;
 -        }
 +        if (this._modalService.isModalOpen) return;

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/24#issuecomment-2514653600>#6ccb3d</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged